File: Medium/1930. Unique Length-3 Palindromic Subsequences/Solution.java
class Solution {
    public int countPalindromicSubsequence(String s) {
        Set<Character> st = new HashSet<>();
        int n = s.length(), ans = 0;
        
        for (int i = 0; i < n; i++) {
            if(!st.contains(s.charAt(i))) {
                st.add(s.charAt(i));
                int lastIdx = s.lastIndexOf(s.charAt(i));
                if (lastIdx != i) {
                    Set<Character> cntDistinct = new HashSet<>();
                    for (int j = i + 1; j < lastIdx; j++) cntDistinct.add(s.charAt(j));
                    ans += cntDistinct.size();
                }
            }
        }
        
        return ans;
    }
}

File: Medium/1930. Unique Length-3 Palindromic Subsequences/Solution.js
var countPalindromicSubsequence = function(s) {
  let st = new Set();
  let n = s.length, ans = 0;

  for(let i = 0; i < n; i++) {
      if(!st.has(s[i])) {
          st.add(s[i]);
          let lastIdx = s.lastIndexOf(s[i]);
          if(lastIdx != i) {
              let cntDistinct = new Set(s.slice(i + 1, lastIdx));
              ans += cntDistinct.size;
          }
      }
  }

  return ans;
};
